src/planet-a/SubdivTriangleMesh.h

```diff
@@ -0,0 +1,487 @@
+
+     , m_chunkIndxFanOffset{ uint32_t(m_chunkWidth)*m_chunkWidth
+                             - smc_minFansVsLevel[subdivLevels] }
+     , m_chunkIndxFillCount{ m_chunkIndxFanOffset + fanMax }
+
+     , m_sharedMax{ sharedMax }
+     , m_sharedIds{ sharedMax }
+     , m_sharedRefCount{ sharedMax }
+     , m_sharedSkVrtx{ std::make_unique<SkVrtxStorage_t[]>(m_sharedMax) }
+     , m_sharedFaceCount{ std::make_unique<uint8_t[]>(m_sharedMax) }
+
+     , m_vrtxSharedOffset{ uint32_t(chunkMax)*m_chunkVrtxFillCount }
+
+     , m_vrtxCountMax{ uint32_t(chunkMax)*m_chunkVrtxFillCount + sharedMax }
+     , m_indxCountMax{ uint32_t(chunkMax) * uint32_t(m_chunkIndxFillCount) }
+    { };
+
+    ChunkId chunk_create(
+            SubdivTriangleSkeleton& rSkel,
+            SkTriId skTri,
+            ArrayView_t<SkVrtxId> const edgeRte,
+            ArrayView_t<SkVrtxId> const edgeBtm,
+            ArrayView_t<SkVrtxId> const edgeLft);
+
+    constexpr VertexId chunk_coord_to_vrtx(
+            ChunkId chunkId, uint16_t x, uint16_t y) const noexcept
+    {
+        if (auto const [localId, shared] = coord_to_shared(x, y, m_chunkWidth);
+            shared)
+        {
+            return shared_get_vrtx(chunk_shared(chunkId)[size_t(localId)]);
+        }
+
+        // Center, non-shared chunk vertex
+        return VertexId(m_chunkVrtxFillCount * size_t(chunkId) + xy_to_triangular(x - 1, y - 2));
+    }
+
+    /**
+     * @return Number of triangles along the edge of a chunk
+     */
+    constexpr uint32_t chunk_width() const noexcept { return m_chunkWidth; }
+
+    /**
+     * @return Number of fill vertices per chunk
+     */
+    constexpr uint32_t chunk_vrtx_fill_count() const noexcept { return m_chunkVrtxFillCount; }
+
+    /**
+     * @brief Get shared vertices used by a specific chunk
+     *
+     * @param chunkId [in] Chunk Id
+     *
+     * @return Array view of shared vertices, access using ChunkLocalSharedId
+     */
+    ArrayView_t<SharedVrtxStorage_t const> chunk_shared(ChunkId chunkId) const noexcept
+    {
+        size_t const offset = size_t(chunkId) * m_chunkVrtxSharedCount;
+        return {&m_chunkSharedUsed[offset], m_chunkVrtxSharedCount};
+    };
+
+    /**
+     *
+     * Param 0: Newly added shared vertices; iterate this.
+     * Param 1: Maps SharedVrtxId to their associated SkVrtxId.
+     * Param 2: The index of the first shared vertex in the vertex buffer.
+     */
+    template<typename FUNC_T>
+    void shared_update(FUNC_T func)
+    {
+        func(ArrayView_t<SharedVrtxId const>(m_sharedNewlyAdded),
+             ArrayView_t<SkVrtxStorage_t const>(m_sharedSkVrtx.get(), m_sharedMax));
+
+        m_sharedNewlyAdded.clear();
+    }
+
+    /**
+     * @return Max number of shared vertices
+     */
+    constexpr uint32_t shared_count_max() const noexcept { return m_sharedMax; }
+
+    uint8_t& shared_face_count(SharedVrtxId sharedId) const noexcept { return m_sharedFaceCount[size_t(sharedId)]; }
```

Comment:
const member functions shouldn't return non-const references to data that's logically owned by this object.